### PR TITLE
Backfill sectionGroupName

### LINKED ISSUE #793: [Course Onboarding] Backfill sectionGroupName
It might be good to write a backfill to set the sectionGroupName of older courses as the same as the course name. Then we can get rid of the nullable constraint on the column.

### PR DESCRIPTION
**Pull request by sumitde22**
# Description

Wrote a quick backfill to fill in the sectionGroupName column in course_model  for old cases where it's null

Allows that column to be marked non-nullable

Closes #793

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [] 
# How Has This Been Tested?

Please describe how you tested this PR (both manually and with tests)
Provide instructions so we can reproduce. 

- [] Test A: Ran backfill locally through yarn cli and saw that dev table was updated correctly. 

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code where needed 
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/server/src/backfill/backfill-section-group-name.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      'set the sectionGroupName of older courses as the same as the course name',
+    autoExit: true,
+  })
+  async fix(): Promise<void> {
+    await CourseModel.createQueryBuilder()
+      .update()
+      .set({ sectionGroupName: () => 'name' })
+      .where('sectionGroupName IS NULL')
+      .execute();
+  }
```
Comment (VKong6019) [inline packages/server/src/backfill/backfill-section-group-name.ts:19]: should we include a log message here to indicate the update occurred? i think logging how many courses were updated is good enough or smth to that effect

**Review by VKong6019 — COMMENTED** [review]
i think other than smol comment. LGTM

File: packages/server/src/backfill/backfill-section-group-name.ts (hunk 1 above)
Comment (sumitde22) [reply packages/server/src/backfill/backfill-section-group-name.ts:20]: Good idea, added

**Review by VKong6019 — APPROVED** [review]
Amazing

**Review by dfarooq610 — APPROVED** [review]
🎉

**Pull request merged**